task/github-close-issue/0.1/README.md

```diff
@@ -1,30 +1,29 @@
-# GitHub
+# Close an issue or a pull request
 
-A collection of tasks to help working with the [GitHub
-API](https://developer.github.com/v3/).
+The `github-close-issue` task let you close a pull request or an
```

Comment:
@chmouel can we change the name of the task, as this task can close an issue or a pull request.
But from the task name`github-close-issue` it seems we can only close issues.

Reply:
I am pretty sure that API endpoint work for issues and pull request.. Since I have used it like that  previously.

Reply:
Yeah it works fine for both issue and pull request, I just mean to say may be we can only change the name of the task so that user can get to know that this task can be used to close issues as well as pull request.

Presently if user reads the name of the task, it would appear to him this task is only to close issue not pull request.

Reply:
Yes but what do you want to call it ?

Reply:
I guess changing the name of the task would be a breaking change and instead of bumping the new version of the task we'll have to add a new task instead which I guess would not be appropriate as there would be two tasks in catalog performing the same work just the name is different

Reply:
Agreed with @vinamra28 , I think now it's not good to rename the task.

Reply:
IMO, one task to one directoy is a bit too cumbersome, if we were going wild we could end up with hundred of github- tasks to implement just one simple call.

Why is is that the hub make it a requirement ?

Could we not have github/close-pull-request/0.1 task/github/close-issue/0.1 etc.. ?

Reply:
@chmouel the current constraint `<kind>/<name>/<version>/<name>.yaml` prevents name collision. since a system/tool can validate that the content of the `name.yaml` has 

```yaml

kind: <kind>
metadata:  
  name: <name>
  labels: 
    app.kubernetes.io/version: <version>
```

In your proposal 
>  Could we not have github/close-pull-request/0.1 task/github/close-issue/0.1 etc.. ?

Any thoughts how you would avoid a name collision and enforce these constraints ? 
* no name collision
* has a version
* The path of resource is computable . i.e we can compute that path `<kind>/<name>/<version>/<name>.yaml` from the metadata. E.g. a tool can compute the location of a `task` named `git-cli` that has a version `0.1`